templates/detail.jinja2

```diff
@@ -0,0 +1,20 @@
+{% extends "base.jinja2" %}
+{% block section %}
+    {% for entry in entries %}
+      {% if not request.authenticated_userid %}
+        <h1>Please Login</h1>
```

Comment:
How many times does this appear if I'm not logged in?

And why are you protecting the details of your posts from un-authenticated users?